lib/y2lan_restart_common.pm

```diff
@@ -94,13 +94,13 @@ sub check_network_status {
 }
 
 sub verify_network_configuration {
-    my ($fn, $dev_name, $expected_status, $workaround) = @_;
+    my ($fn, $dev_name, $expected_status, $workaround, $no_network_check) = @_;
     open_network_settings;
 
     $fn->($dev_name) if $fn;                                          # verify specific action
 
     close_network_settings;
-    check_network_status($expected_status, $workaround);
+    check_network_status($expected_status, $workaround) unless defined $no_network_check;
```

Comment:
Yes, exactly. Seems that I got distracted while typing =)

Reply:
This works but I would need to refactor other functions to make it coherent, otherwise I find it confusing rather than helping. "my ($args) = @_;" works, but to use the function "verify_network_configuration" I need to put the arguments in order e.g `verify_network_configuration(\&set_dhcp, undef, 'restart');` or `verify_network_configuration(sub { set_static_ip($looprun) }, undef, undef, undef, 'no_network_check');` . If I remove the "undefs"  here (e.g  `verify_network_configuration(\&set_dhcp, 'restart' );` ) it does not work. So I kept it as is to make it clear how many arguments we need to specify when we use the function.

Reply:
It's not exactly what I proposed, I was talking about using hash here and not getting array reference, which would be non-sense. If hash is used (doesn't matter if reference or hash directly) and then you can access key which you need and pass only those which you need.
So you will parse arguments like
```perl
sub aaa {
my(%args)=@_; 
print $args{param1};
}

aaa (param1 => 1, param2 =>2);
```
Hope now it's clear. It was optional suggestion, but we definitely do not need to check if parameter is defined or not, but use boolean logic of perl.